File: RoadTrip/rt.rb
# road trip soln in rb for codeeval by steven a dunn

File.open(ARGV[0], 'r').each_line do |line|
    line = line.chomp.split(";")
    distances = []
    line.each do |entry|
        distances << entry.split(",")[1]
    end
    distances.map!(&:to_i).sort!
    results = []
    cur_pos = 0
    distances.each do |distance|
        location = distance - cur_pos
        results << location
        cur_pos = distance
    end
    puts results.join(",")
end

File: RoadTrip/py2/rt.py
# road trip soln in py2 for codeeval by steven a dunn

import sys

f = open(sys.argv[1], 'r')
for line in f:
    line = line.rstrip().split(";")
    distances = []
    for entry in line:
        if not entry:
            continue
        distance = int(entry.split(",")[1])
        distances.append(distance)
    distances.sort()

    results = []
    cur_pos = 0
    for distance in distances:
        location = distance - cur_pos
        results.append(location)
        cur_pos = distance
    results = list(map(str, results))
    print ",".join(results)
f.close()